queue_services/entity-emailer/src/entity_emailer/email_processors/incorp_notification.py

```diff
@@ -12,13 +12,166 @@
+        else:
+            noa_encoded = base64.b64encode(noa.content)
+            pdfs.append(
+                {
+                    'fileName': 'Notice of Articles.pdf',
+                    'fileBytes': noa_encoded.decode('utf-8'),
+                    'fileUrl': '',
+                    'attachOrder': '1'
+                }
+            )
+        certificate = requests.get(
+            f'{current_app.config.get("LEGAL_API_URL")}/businesses/{business["identifier"]}/filings/{filing.id}\
+            ?type=certificate',
+            headers=headers
+        )
+        if certificate.status_code != HTTPStatus.CREATED:
+            logger.error('Failed to get certificate pdf for filing: %s', filing.id)
+            capture_message(f'Email Queue: filing id={filing.id}, error=certificate generation', level='error')
+        else:
+            certificate_encoded = base64.b64encode(certificate.content)
+            pdfs.append(
+                {
+                    'fileName': 'Incorporation Certificate.pdf',
+                    'fileBytes': certificate_encoded.decode('utf-8'),
+                    'fileUrl': '',
+                    'attachOrder': '2'
+                }
+            )
 
+    return pdfs
 
-def process(email_msg):
+
+def _get_recipients(stage: str, filing_json: dict):
+    """Get the recipients for the incorporation output."""
+    recipients = filing_json['filing']['incorporationApplication']['contactPoint']['email']
+    if stage == 'filed':
+        parties = filing_json['filing']['incorporationApplication'].get('parties')
+        comp_party_email = None
+        for party in parties:
+            for role in party['roles']:
+                if role['roleType'] == 'Completing Party':
+                    comp_party_email = party['officer']['email']
+                    break
+        recipients = f'{recipients}, {comp_party_email}'
+    return recipients
+
+
+def process(email_msg: dict, token: str):
     """Build the email for Business Number notification."""
     logger.debug('incorp_notification: %s', email_msg)
-    email = 'incorp hardcoded'
-    # build email from template
-    # get links etc.
-    return email
+    # get template and fill in parts
+    template = Path(f'email_templates/BC-{email_msg["option"]}-success.html').read_text()
+    filled_template = substitute_template_parts(template)
+
+    # get template vars from filing
+    filing = Filing.find_by_id(email_msg['filingId'])
+    filing_json = filing.json
+    business = filing_json['filing']['business']
+    filing_date = datetime.fromisoformat(filing.filing_date.isoformat())
+    leg_tmz_filing_date = LegislationDatetime.as_legislation_timezone(filing_date).strftime('%Y-%m-%d %I:%M %p')
+    effective_date = datetime.fromisoformat(filing.effective_date.isoformat())
+    leg_tmz_effective_date = LegislationDatetime.as_legislation_timezone(effective_date).strftime('%Y-%m-%d %I:%M %p')
```

Comment:
This does not appear consistent with the report date formats, eg
`filing['filing_date_time'] = filing_datetime.strftime(f'%B %d, %Y {hour}:%M %p Pacific Time')`

Reply:
You mean the report dates in the pdfs? Yeah idk what the format is supposed to be. The dates aren't in the google doc and I haven't seen a UI design or anything

Reply:
they don't list the dates in the google doc either so are we sure these dates are even staying in?